bin/journey-test.sh

```diff
@@ -215,12 +233,12 @@ solve_all_exercises() {
 
     pushd ${exercism_exercises_dir}/${TRACK}/${exercise}
     # Check that tests compile before we strip @Ignore annotations
-    gradle compileTestJava
+    "$EXECPATH"/gradlew compileTestKotlin
     # Ensure we run all the tests (as delivered, all but the first is @Ignore'd)
     for testfile in `find . -name "*Test.${TRACK_SRC_EXT}"`; do
-      sed 's/@Ignore//' ${testfile} > "${tempfile}" && mv "${tempfile}" "${testfile}"
+      sed 's/@Ignore\(\(.*\)\)\{0,1\}//' ${testfile} > "${tempfile}" && mv "${tempfile}" "${testfile}"
```

Comment:
Ok, so I think I understand this now:

- if we are running all tests from the 'unit' perspective, this copy-and-filter task is what creates a copy of the tests minus any ignore annotations in the `build/gen/test/java` directory, which is also specified as the location of the test source set;
- if we are running all tests from the 'journey' perspective, there's no 'root' project any more, so in order to simulate an implementor running all the tests, we need to manually strip ignore annotations in place.

Unfortunately the only way to avoid duplicating the 'strip ignore annotations' parts of these cases would appear to be packaging a copy task with the Gradle files that are served to implementors, which isn't great. As an alternative, maybe we should insert comments in both locations to help us remember that the two 'strip' implementations should be analogous. I'm thinking that's preferable.